### ISSUE TITLE
cocoapods script phases

### ISSUE THREAD

**Issue by idkjs** [react-native-config/react-native-config#514]
Upon installing this package and running `npx pod-install` I get this message:

```sh
[!] react-native-config has added 1 script phase. Please inspect before executing a build. See `https://guides.cocoapods.org/syntax/podspec.html#script_phases` for more information.
```

Does this require us do anything?

Previously, I set the config variables in a file. Are env variables now required?

Thank you.

**Comment by kockok** [comment]
I got this error after executing `pod install` and then XCode Build Failed and raising `PhaseScriptExecution failed with a nonzero exit code` error.

**Comment by muhammetvrl** [comment]
> I get the same problem

**Comment by syedquadri1982** [comment]
is this resolved yet?

**Comment by esteban1213** [comment]
I will pay to whoever solves this

**Comment by OmarSukarieh** [comment]
for me i got this warning and all i do is re-run "pod install" in ios folder

**Comment by esteban1213** [comment]
I have done all the basic procedures like reinstalling and deleting stuff and running things over and over. you know the usual "might solve it" stuff. I am ready to quit my entire career

**Comment by OmarSukarieh** [comment]
i dont know but maybe you need to update ur XCode, i only got this error when i delete pods files and reinstall pods but i run pod install againg and its working now just fine : )

**Comment by esteban1213** [comment]
Yeah when I run pod install again, the error doesn't show anymore, but the build still has this error.
Command PhaseScriptExecution failed with a nonzero exit code

**Comment by MauroAlberto1992** [comment]
Waiting for the magical fix!
Maybe some update burned the xcode

**Comment by longb1997** [comment]
+1

**Comment by rcwestlake** [comment]
Any update on this? I'm getting the same error.

**Comment by brenopgarcia** [comment]
Why God, Why?

**Comment by Nicat26** [comment]
Any update?

**Comment by andrewshortall** [comment]
+1

**Comment by ganganimaulik** [comment]
I found this solution on internet somewhere: 


**step-1:**
search for `[CP-User] Generate Specs` in `ios/Pods/Pods.xcodeproj/project.pbxproj` 

and delete this step ie:
`
17D1E985C9AAF5C7D138FFE6E6201792 /* [CP-User] Generate Specs */ = {
    isa = PBXShellScriptBuildPhase;
    buildActionMask = 2147483647;
    files = (
    );
    inputPaths = (
        "/Users/maulik/Desktop/audioboom-app-main/node_modules/react-native/scripts/../Libraries",
    );
    name = "[CP-User] Generate Specs";
    outputPaths = (
        "$(DERIVED_FILE_DIR)/codegen-FBReactNativeSpec.log",
        "/Users/maulik/Desktop/audioboom-app-main/node_modules/react-native/scripts/../React/FBReactNativeSpec/FBReactNativeSpec/FBReactNativeSpec.h",
        "/Users/maulik/Desktop/audioboom-app-main/node_modules/react-native/scripts/../React/FBReactNativeSpec/FBReactNativeSpec/FBReactNativeSpec-generated.mm",
    );
    runOnlyForDeploymentPostprocessing = 0;
    shellPath = /bin/sh;
    shellScript = "set -o pipefail\n\nbash -l -c 'SRCS_DIR=/Users/maulik/Desktop/audioboom-app-main/node_modules/react-native/scripts/../Libraries CODEGEN_MODULES_OUTPUT_DIR=/Users/maulik/Desktop/audioboom-app-main/node_modules/react-native/scripts/../React/FBReactNativeSpec/FBReactNativeSpec CODEGEN_MODULES_LIBRARY_NAME=FBReactNativeSpec /Users/maulik/Desktop/audioboom-app-main/node_modules/react-native/scripts/generate-specs.sh' 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"";
};
`

**step-2**
run `./node_modules/react-native/scripts/generate-specs.sh`

**step-3**
build

**Comment by erazavr** [comment]
has anyone been able to solve this issue?

**Comment by brenopgarcia** [comment]
I've changed the react-native version to "0.63.2", delete node-modules folder, and run yarn. It works for me.

**Comment by sjain0700** [comment]
I resolved this issue  following these steps.
1. Updated the xcode 12.5  and uninstall previous xcode vesions(if installed)
2. update macOS version 11.4 
3.  cd ios
pod deintegrate
cd ..
rm -rf ios/Podfile.lock
yarn install or npm install
yarn audit fix
cd ios
pod install
cd ..
react-native run-ios  , (if Stii not run then open project file .xcworkspace in xcode 12.5 and run)

**Comment by Yadurshanm** [comment]
This worked for me https://github.com/react-native-community/upgrade-support/issues/138#issuecomment-808895860

The spaces in the folder path was the issue

**Comment by lazharkarem** [comment]
should Delete alias default  :
** run** ( "nvm unalias default"  )

**Comment by SteveHan-233** [comment]
I'm having this issue in EAS build

**Comment by renan-cardoso** [comment]
> I will pay to whoever solves [!] FBReactNativeSpec has added 1 script phase. Please inspect before executing a build. See `https://guides.cocoapods.org/syntax/podspec.html#script_phases` for more information.

What worked for me was to run `export DISABLE_CODEGEN=1` and then run `npx pod install` on the terminal.

the file `/node_modules/react_native/React/FBReactNativeSpec/FBReactNativeSpec.podspec` have a function that doesn't add a script_phase to podspec if there is an ENV variable DISABLE_CODEGEN igual to 1.

**Comment by Kabeer-Haseja** [comment]
+1

**Comment by Bardashevsky** [comment]
Helped for me:
pod deintegrate
pod install  

After that will be an error :
... has added 2 script phases. Please inspect before executing a build. See `https://guides.cocoapods.org/syntax/podspec.html#script_phases` for more information.

Then do again with out pod deintegrate:
pod install  
or 
bundle exec pod install
